packages/clay-form/src/MultiSelect.tsx

```diff
@@ -0,0 +1,314 @@
+/**
+ * © 2019 Liferay, Inc. <https://liferay.com>
+ *
+ * SPDX-License-Identifier: BSD-3-Clause
+ */
+
+import classNames from 'classnames';
+import ClayAutocomplete from '@clayui/autocomplete';
+import ClayIcon from '@clayui/icon';
+import ClayLabel from '@clayui/label';
+import React, {useLayoutEffect, useRef, useState} from 'react';
+
+const COMMA = 188;
+const ENTER = 13;
+
+const DELIMITER_KEYS = [ENTER, COMMA];
+
+const BACKSPACE_KEY = 8;
+
+export interface IProps extends React.HTMLAttributes<HTMLInputElement> {
+	/**
+	 * Message to display if validation fails.
+	 */
+	errorMessage?: string;
+
+	/**
+	 * Ref added to container element
+	 */
+	forwardRef: React.RefObject<HTMLDivElement>;
+
+	/**
+	 * Message to display which helps users interact with input
+	 */
+	helpText?: string;
+
+	/**
+	 * Value of input
+	 */
+	inputValue?: string;
+
+	/**
+	 * Values that display as label items
+	 */
+	items: string[];
+
+	/**
+	 * Description of the input
+	 */
+	label?: React.ReactText;
+
+	/**
+	 * Callback for when items are added or removed
+	 */
+	onItemsChange: (val: string[]) => void;
+
+	/**
+	 * Callback for when the input value changes
+	 */
+	onInputChange?: (val: string) => void;
+
+	/**
+	 * List of pre-populated items that will show up in a dropdown menu
+	 */
+	sourceItems?: string[];
+
+	/**
+	 * Path to spritemap for clay icons
+	 */
+	spritemap?: string;
+
+	/**
+	 * Flag to indicate if validation should occur when input is blured
+	 */
+	validateOnBlur?: boolean;
+
+	/**
+	 * Callback for validating item. Returning `true` means the value is valid.
+	 */
+	validator?: (val?: string) => boolean;
+}
+
+const MultiSelect: React.FunctionComponent<IProps> = ({
+	errorMessage,
+	forwardRef,
+	helpText,
+	items,
+	label,
+	onItemsChange,
+	spritemap,
+	validateOnBlur = false,
+	validator = () => true,
+	inputValue = '',
+	onInputChange,
+	sourceItems,
+	...otherProps
+}) => {
+	const inputRef = useRef<HTMLInputElement>(null);
+	const lastItemRef = useRef<HTMLSpanElement>(null);
+	const [active, setActive] = useState(false);
+	const [isFocused, setIsFocused] = useState();
+	const [isValid, setIsValid] = useState(true);
+
+	useLayoutEffect(() => {
+		if (sourceItems) {
+			const matchedItems = sourceItems.filter(
+				item => inputValue && item.match(inputValue)
+			);
+
+			setActive(matchedItems.length !== 0);
+		}
+	}, [sourceItems, inputValue]);
+
+	const setNewValue = (newVal: string) => {
+		if (validator(newVal)) {
+			onItemsChange([...items, newVal]);
```

Comment:
Another use case too, is that we have to make sure not to label a comma. For example, try pressing comma twice.